core/src/codec.rs

```diff
@@ -44,6 +55,18 @@ pub trait Decode<C: Codec>: Sized {
     fn decode<R: Read>(c: C, r: &mut R) -> Result<Self>;
 }
 
+/// References trait.
+///
+/// This trait is generic over a codec, so that different codecs can be implemented for the same
+/// type.
+pub trait References<C: Codec>: Sized {
+    /// Scrape the references from an `impl Read`.
+    ///
+    /// It takes a specific codec as parameter, so that the [`References`] can be generic over an
+    /// enum that contains multiple codecs.
+    fn references<R: Read + Seek>(c: C, r: &mut R, set: &mut FnvHashSet<Cid>) -> Result<()>;
```

Comment:
Could we avoid having to have a dependency on fnv by having the caller decide what collection to use for aggregation? Not the most important thing in the world, but some callers might want a different collection.

Maybe leave the set type generic, but just require https://doc.rust-lang.org/std/iter/trait.Extend.html ?